lib/spack/spack/util/compression.py

```diff
@@ -49,9 +83,9 @@ def _unzip(archive_file):
         archive_file (str): absolute path of the file to be decompressed
     """
     try:
-        from zipfile import ZipFile
+        SpackZip = spack_unzipper()
         destination_abspath = os.getcwd()
-        with ZipFile(archive_file, 'r') as zf:
+        with SpackZip(archive_file, 'r') as zf:
```

Comment:
I missed it in earlier review but there is a conceptual mismatch in terms of the return value of the function: we either return a `callable` object, or we extract the archive file and return `None`, we should be consistent about that.

Reply:
Agreed. I'll have this return None on all platforms as a callable ZipFile would require passing a function pointer to `ZipFile.extractall`, and I'd rather keep control over that process on the compression side. If the convention is a callable, that can work as well though. 

Note: the compression PR overhauls a large amount of this, and would be useful to have in this scenario.

Reply:
I'm surprised systems without `ZipFile` available weren't failing on the returned unzip callable.